message-index/messages/GHC-97739/example/after/DependencyOrder.hs

```diff
@@ -0,0 +1,9 @@
+{-# LANGUAGE DataKinds, PolyKinds, ExplicitForAll #-}
+module DependencyOrder where
+
+import Data.Kind
+
+data SameKind :: k -> k -> *
+
+foo :: forall k a (b :: k). SameKind a b
+foo = undefined
```

Comment:
Did we want to avoid `undefined` if possible? Or is there no reasonable value for foo in this case?